nsv13/code/modules/coffee/machine/impressa.dm

```diff
@@ -0,0 +1,243 @@
+ */
+
+/obj/machinery/coffeemaker/impressa
+	name = "impressa coffeemaker"
+	desc = "An industry-grade Impressa Modello 5 Coffeemaker of the Piccionaia Home Appliances premium coffeemakers product line. Makes coffee from fresh dried whole beans."
+	icon = 'nsv13/icons/obj/coffee.dmi'
+	icon_state = "coffeemaker_impressa"
+	circuit = /obj/item/circuitboard/machine/coffeemaker/impressa
+	initial_cartridge = null		//no cartridge, just coffee beans
+	brew_time = 15 SECONDS			//industrial grade, its faster than the regular one
+	density = TRUE
+	pass_flags = PASSTABLE
+
+	var/coffee_amount = 0
+	//this type of coffeemaker takes fresh whole beans insted of cartidges
+	var/list/coffee = list()
+
+/obj/machinery/coffeemaker/impressa/Initialize(mapload)
+	. = ..()
+	if(mapload)
+		coffeepot = new /obj/item/reagent_containers/glass/coffeepot(src)
+		cartridge = null
+
+
+/obj/machinery/coffeemaker/impressa/Destroy()
+	QDEL_NULL(coffeepot)
+	QDEL_NULL(coffee)
+	return ..()
+
+/obj/machinery/coffeemaker/impressa/Exited(atom/movable/gone, direction)
+	if(gone == coffeepot)
+		coffeepot = null
+	if(gone == coffee)
+		coffee = null
+	return ..()
+
+/obj/machinery/coffeemaker/impressa/examine(mob/user)
+	. = ..()
+	if(coffee)
+		. += "<span class='notice'The internal grinder conatins [coffee.len] scoop\s of coffee beans.</span>"
+	return
+
+/obj/machinery/coffeemaker/impressa/update_overlays()
+	. = ..()
+	. += overlay_checks()
+
+/obj/machinery/coffeemaker/impressa/overlay_checks()
+	. = list()
+	if(coffeepot)
+		if(coffeepot.reagents.total_volume > 0)
+			. += "pot_full"
+		else
+			. += "pot_empty"
+	if(coffee_cups > 0)
+		if(coffee_cups >= max_coffee_cups/3)
+			if(coffee_cups > max_coffee_cups/1.5)
+				. += "cups_3"
+			else
+				. += "cups_2"
+		else
+			. += "cups_1"
+	if(sugar_packs)
+		. += "extras_1"
+	if(creamer_packs)
+		. += "extras_2"
+	if(sweetener_packs)
+		. += "extras_3"
+	if(coffee_amount)
+		if(coffee_amount < 0.7*BEAN_CAPACITY)
+			. += "grinder_half"
+		else
+			. += "grinder_full"
+	return
+
+/obj/machinery/coffeemaker/impressa/handle_atom_del(atom/A)
+	. = ..()
+	if(A == coffeepot)
+		coffeepot = null
+	if(A == coffee)
+		coffee.Cut()
+	update_icon()
+
+/obj/machinery/coffeemaker/impressa/can_brew()
+	if(coffee_amount <= 0)
+		balloon_alert_to_viewers("no coffee beans added!")
+		return FALSE
+	if(!coffeepot)
+		balloon_alert_to_viewers("no coffeepot inside!")
+		return FALSE
+	if(machine_stat & (NOPOWER|BROKEN) )
+		balloon_alert_to_viewers("machine unpowered!")
+		return FALSE
+	if(coffeepot.reagents.total_volume >= coffeepot.reagents.maximum_volume)
+		balloon_alert_to_viewers("the coffeepot is already full!")
+		return FALSE
+	return TRUE
+
+/obj/machinery/coffeemaker/impressa/attackby(obj/item/attack_item, mob/user, params)
+	//You can only screw open empty coffeemakers
+	if(!coffeepot && default_deconstruction_screwdriver(user, icon_state, icon_state, attack_item))
+		return
```

Comment:
Do these need return values since all the rest of this has them?

Reply:
Unsure, I believe that is there due to the default_deconstruction_screwdriver proc being called